features/202-session-management/config.md

```diff
@@ -0,0 +1,57 @@
+# Session Configuration
+
+## Sample configuration
+
+In user configuration:
+```yaml
+clients:
```

Comment:
Evil idea: `session_statefulness: stateless` XD

Reply:
Stateless session is not desirable under current design:
- Each request would need to read user information from store (maybe PgSQL/Redis), so using stateless session does not avoid DB hit.
- A list of revoked session ID need to be maintained, so stateless session would not be really stateless.
- We had decided to drop client-side token introspection support for now (i.e. JWT would not be used).

Reply:
I don't agree it is not desirable... actually I think some how in future we would do it...

>     * Each request would need to read user information from store (maybe PgSQL/Redis), so using stateless session does not avoid DB hit.

Yes but IO is IO, one less IO would have shorter latency. Plus if we design to be stateless maybe we can optimize it to a point where user information are in the stateless sessions (so larger network payload in exchange for lower IO overhead and easier to scale)



>     * A list of revoked session ID need to be maintained, so stateless session would not be really stateless.

We can drop support of terminate sessions for stateless sessions



>     * We had decided to drop client-side token introspection support for now (i.e. JWT would not be used).

"for now" :)

Reply:
As mentioned, it is undesirable 'under current design'. Of course we can revisit it in future, but the current design does not support truly stateless session.

Reply:
So I think we can create a feature issue about it? And mention it is likely we can add a `session_statefulness` option in this configuration?

Reply:
BTW, @kiootic when I said an "evil idea", obviously I don't expect we would have it in current design.

Reply:
opened #353 to track.